### PR TITLE
add todo form solution

### PR DESCRIPTION
**Pull request by vladyslav-harkusha**
[DEMO LINK](https://vladyslav-harkusha.github.io/react_add-todo-form/)

### CODE REVIEW

**Review by Anna-Borodaienko — APPROVED** [review]

File: src/react-app-env.d.ts [hunk 1]
```diff
@@ -1 +1,19 @@
...
+export interface Todo {
+  id: number;
+  title: string;
+  completed: boolean;
+  userId: number;
+}
+
+export interface TodoWithUser extends Todo {
+  user: User | null;
+}
```
Comment (sergii-nosachenko) [inline src/react-app-env.d.ts:19]:
```suggestion
```
Move it to separate files in `types` folder.

File: src/App.tsx [hunk 2]
```diff
@@ -1,61 +1,125 @@
...
         <div className="field">
-          <select data-cy="userSelect">
+          <select
+            data-cy="userSelect"
+            value={todoUserId}
+            onChange={handleSelect}
+          >
             <option value="0" disabled>Choose a user</option>
+
+            {usersFromServer.map(user => (
```
Comment (sergii-nosachenko) [inline src/App.tsx:105]: Can be destructurized.

File: src/App.tsx [hunk 3]
```diff
@@ -1,61 +1,125 @@
...
 
-      <form action="/api/users" method="POST">
+      <form
+        method="POST"
+        onSubmit={handleFormSubmit}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <input
```
Comment (sergii-nosachenko) [inline src/App.tsx:84]: Add field label

File: src/App.tsx [hunk 4]
```diff
@@ -1,61 +1,125 @@
...
+          />
+
+          {titleError && (
+            <span className="error">Please enter a title</span>
+          )}
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
+          <select
```
Comment (sergii-nosachenko) [inline src/App.tsx:98]: Add field label

File: src/App.tsx [hunk 5]
```diff
@@ -1,61 +1,125 @@
...
+
+    const newTodo = {
+      id: getCorrectId(todos),
+      title: todoTitle,
+      completed: false,
+      userId: todoUserId,
+      user: getUserById(todoUserId),
+    };
+
+    if (!titleError && !userIdError) {
```
Comment (sergii-nosachenko) [inline src/App.tsx:58]:
```suggestion
```
You have checked this condition on line 46

**Review by sergii-nosachenko — CHANGES_REQUESTED** [review]

**Review by sergii-nosachenko — APPROVED** [review]
Good work 🚀🚀🚀

**Pull request closed without merging**